paddle/fluid/operators/controlflow/while_op.cc

```diff
@@ -341,6 +365,52 @@ class WhileGradOpDescMaker : public framework::SingleGradOpDescMaker {
     // while operator could be renamed.
     while_grad->SetAttr("original_output_grad", output_grads_list);
 
+    /* The following codes are used in eager deletion mode */
+    std::unordered_set<std::string> bwd_skip_vars;
```

Comment:
Program Optimization logic shouldn't leak into the operator. Otherwise, all operators can become super complex and filled with various global optimization logics.
This logic should be implemented in Prepare() or Pass.

Reply:
`while_op` is quite special. Current `ir::Graph` is a simple DAG without nested structure. `while_op` inside this DAG is a simple op as other ops. But `while_op` differs from other ops in two ways:
- `while_op` has a nested structure, which `ir::Graph` cannot represent now.
- `while_grad_op` and `while_op` resides in two different blocks, meaning that two different nested DAGs should be considered together, which `ir::Graph` cannot represent too.

BTW, maybe it is not a better choice to implement this logic inside `Prepare`. If it is inside `Prepare`, we should add codes to distinguish whether the running op is `while_op` `while_gra_op` or other ops, which is not readable and elegant.